flexmeasures/data/schemas/scheduling/__init__.py

```diff
@@ -31,3 +32,52 @@ class FlexContextSchema(Schema):
     inflexible_device_sensors = fields.List(
         SensorIdField(), data_key="inflexible-device-sensors"
     )
+
+
+class SequentialFlexModelSchema(Schema):
+    sensor = SensorIdField(required=True)
+    sensor_flex_model = fields.Dict(data_key="sensor-flex-model")
+
+    @pre_load
+    def unwrap_envelope(self, data, **kwargs):
+        """Any field other than 'sensor' becomes part of the sensor's flex-model."""
+        extra = {}
+        rest = {}
+        for k, v in data.items():
+            if k not in self.fields:
+                extra[k] = v
+            else:
+                rest[k] = v
+        return {"sensor-flex-model": extra, **rest}
+
+
+class AssetTriggerSchema(Schema):
+    asset = GenericAssetIdField(data_key="id", status_if_not_found=404)
```

Comment:
This schema could be used via de CLI so the `status_if_not_found` will not make sense there. As an alternative, we can catch the exception in the CLI and display the proper message.

Reply:
Alternatively, we can avoid this situation by providing an already deserialized asset using the `AssetIdField` as a CLI field.

Reply:
Removing the `status_if_not_found=404` here would mean that visiting `/assets/<non-existing-id>/schedules/trigger` would return a validation error (422 status?) rather than 404 Not Found.

I would actually be fine with that. Given that we are POSTing a resource (job) rather than GETting a resource (asset) here, the `asset` should be seen as part of the POSTed payload to be validated.
Even though at first glance it doesn't seem correct if the asset ID is coming from the path, we have already started to treat the payload more RESTlike by including it in with the JSON data to be validated within the same schema.
In the future, we'd want to move to a RESTful approach to jobs anyway, and then the `asset` truly becomes part of the JSON payload.

We might already reconsider the choice of endpoint, but I am wary of increasing the scope of this PR too much.

For now I'll remove the 404 here.

Concerning the CLI, we hardly use schemas there at the moment, as opposed to fields, so maybe it's not that much of a problem?